packages/remix-react/components.tsx

```diff
@@ -1408,36 +1409,35 @@ export const LiveReload =
          */
         nonce?: string;
       }) {
-        let setupLiveReload = (port: number) => {
-          let protocol = location.protocol === "https:" ? "wss:" : "ws:";
-          let host = location.hostname;
-          let socketPath = `${protocol}//${host}:${port}/socket`;
-
-          let ws = new WebSocket(socketPath);
-          ws.onmessage = (message) => {
-            let event = JSON.parse(message.data);
-            if (event.type === "LOG") {
-              console.log(event.message);
-            }
-            if (event.type === "RELOAD") {
-              console.log("💿 Reloading window ...");
-              window.location.reload();
-            }
-          };
-          ws.onerror = (error) => {
-            console.log("Remix dev asset server web socket error:");
-            console.error(error);
-          };
-        };
-
-        React.useEffect(() => {
-          if (!liveReloadMounted) {
-            setupLiveReload(port);
-            liveReloadMounted = true;
-          }
-        }, []);
-
-        return null;
+        return (
+          <script
+            nonce={nonce}
+            suppressHydrationWarning
+            dangerouslySetInnerHTML={{
+              __html: `
+                let protocol = location.protocol === "https:" ? "wss:" : "ws:";
```

Comment:
Are we concerned about using `let` instead of `var` here? It won't be compiled, but maybe we don't care?

Also, I can't remember whether `let` variables will be added to the global scope. I think they won't, but we'll want an iife otherwise.

Reply:
They do cross script boundaries and end up on the global scope, good catch. Just iife'd it.